search.go

```diff
@@ -64,69 +44,169 @@ func (c *SearchController) Show(ctx *app.ShowSearchContext) error {
+		fmt.Println(err)
+		ctx.InternalServerError()
+	}
+
+	var users []*app.Users
+	for i := range result {
+		ident := result[i]
+		id := ident.ID.String()
+		users = append(users, &app.Users{
+			Type: "users",
+			ID:   &id,
+			Attributes: &app.UserAttributes{
+				Fullname: &ident.FullName,
+				ImageURL: &ident.ImageURL,
+			},
+		})
+	}
+
+	pagingLinks := paginationLinks(
+		buildAbsoluteURL(ctx.RequestData),
+		"q="+ctx.Q,
+		offset,
+		limit,
+		count,
+		len(result))
+
+	response := app.SearchResponseUsers{
+		Data:  users,
+		Links: &pagingLinks,
+		Meta:  map[string]interface{}{"total-count": count},
+	}
+
+	return ctx.OK(&response)
+}
+
+func paginationLinks(path, args string, offset, limit, count, length int) app.PagingLinks {
+	links := app.PagingLinks{}
+
+	// prev link
+	if offset > 0 && count > 0 {
+		var prevStart int
+		// we do have a prev link
+		if offset <= count {
+			prevStart = offset - limit
 		} else {
-			// retreat at least one page until covering the range
-			lastStart = offset - ((((offset - count) / limit) + 1) * limit)
+			// the first range that intersects the end of the useful range
+			prevStart = offset - (((offset-count)/limit)+1)*limit
 		}
 		realLimit := limit
-		if lastStart < 0 {
+		if prevStart < 0 {
 			// need to cut the range to start at 0
-			realLimit = limit + lastStart
-			lastStart = 0
+			realLimit = limit + prevStart
+			prevStart = 0
 		}
-		last := fmt.Sprintf("%s?q=%s&page[offset]=%d&page[limit]=%d", buildAbsoluteURL(ctx.RequestData), ctx.Q, lastStart, realLimit)
-		response.Links.Last = &last
+		prev := fmt.Sprintf("%s?%s&page[offset]=%d&page[limit]=%d", path, args, prevStart, realLimit)
+		links.Prev = &prev
+	}
 
-		return ctx.OK(&response)
-	})
+	// next link
+	nextStart := offset + length
+	if nextStart < count {
+		// we have a next link
+		next := fmt.Sprintf("%s?%s&page[offset]=%d&page[limit]=%d", path, args, nextStart, limit)
+		links.Next = &next
+	}
+
+	// first link
+	var firstEnd int
+	if offset > 0 {
+		firstEnd = offset % limit // this is where the second page starts
+	} else {
+		// offset == 0, first == current
+		firstEnd = limit
+	}
+	first := fmt.Sprintf("%s?%s&page[offset]=%d&page[limit]=%d", path, args, 0, firstEnd)
+	links.First = &first
+
+	// last link
+	var lastStart int
+	if offset < count {
+		// advance some pages until touching the end of the range
+		lastStart = offset + (((count - offset - 1) / limit) * limit)
+	} else {
+		// retreat at least one page until covering the range
+		lastStart = offset - ((((offset - count) / limit) + 1) * limit)
+	}
+	realLimit := limit
+	if lastStart < 0 {
+		// need to cut the range to start at 0
+		realLimit = limit + lastStart
+		lastStart = 0
+	}
+	last := fmt.Sprintf("%s?%s&page[offset]=%d&page[limit]=%d", path, args, lastStart, realLimit)
+	links.Last = &last
+
+	return links
+}
+
+// paginationOptions parses and defaults offset/limit numbers
+func paginationOptions(pageOffset *string, pageLimit *int) (int, int) {
```

Comment:
I think it makes sense to name the return values when they are just two ints. Makes it harder to easier to not get the order wrong when calling.